### ISSUE TITLE
Spam paste

### ISSUE THREAD

**Issue by tvvocold** [KennyShields/LightPaste#8]
I just check the datebase and find 2000+ spam paste, i think maybe the api reason, should i delete the api.php or find another way?

**Comment by KennyShields** [comment]
Yeah, I have the same problem at lightpaste.com, there are about 28,000 spam pastes in the database so far. However, the problem doesn't seem to be with the api, as my sitelogs indicate that the spam bot is accessing /new url to create new pastes. One possible solution might be to make use of Fat Free's built in blacklist system: http://fatfreeframework.com/base#blacklisted. I'm not sure how well that will work though, as the bot seems to have hundreds, (possibly thousands) of different ips.

**Comment by tvvocold** [comment]
how about add a simple capcha?

**Comment by KennyShields** [comment]
Thats another possibility, although I want to avoid adding a captcha as it makes using the site less fluid. However, I suppose we could make it optional and have a setting for it somewhere in the config file.

**Comment by tvvocold** [comment]
Sure, check this [recapcha](https://www.google.com/recaptcha/intro/index.html).
or [simple php capcha](https://github.com/search?utf8=%E2%9C%93&q=simple+captcha+php).

**Comment by tvvocold** [comment]
Now i got 20000+ spam...lol

**Comment by KennyShields** [comment]
Just enabled the dns blacklist in commit 5b307541a75a4cad7f84b5a692db0dc90b567ecd. I suggest updating your installation of Light Paste to see if that helps reduce the spam.

**Comment by tvvocold** [comment]
So i need to recreate the datebase? (like change mysql to sqlite, but i cannot delete users data)
So far i had update to snap version.And it works well.

**Comment by KennyShields** [comment]
You can leave your database as is, I'm just suggesting that you update to the version with blacklisting enabled to see if that helps reduce the rate of spam pastes.

**Comment by tvvocold** [comment]
@KennyShields 
Thx & Sry , I need to know does this work in your site?( lightpaste.com) ,then i could do a git pull.

**Comment by KennyShields** [comment]
The most recent version has some db schema changes, so you'd need to add those into your database.

**Comment by tvvocold** [comment]
hi @KennyShields ,
Did you remove the API page,why? (Just found there is 404 in http://lightpaste.com/api )

**Comment by KennyShields** [comment]
Yeah, I removed the information pages a while ago.
